implementations/dotnet-corvus-jsonschema/Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+WORKDIR /source
+
+COPY *.csproj .
+
+COPY . .
+RUN dotnet publish -c Release -o /app
```

Comment:
Interestingly, this doesn't work with a local build for me, so I reverted to a single restore, build, and publish. But I'll have another look now it is functioning.

Reply:
OK - I'm a bit stumped as to why this doesn't build!

Reply:
Oh, hang on - it's building on ARM64 for some reason...

Reply:
Yes. It doesn't all blow up on AMD64, but explodes on ARM64.

Reply:
I'll have to see if I can set aside some more time to play with it myself -- but yeah that bit is expected, we build all implementations in multiarch containers for both AMD64 and ARM64.

Reply:
There's no reason the dotnet code shouldn't run on ARM64 (in fact, we know it runs on modern MacOS machines, so that should be fine.)

It might be something to do with Roslyn on ARM64...

https://github.com/dotnet/roslyn/issues/59421

Reply:
I looked last night here for just a few minutes and my sneaky suspicions led me to a .NET issue saying that QEMU is [not really supported](https://github.com/dotnet/core/pull/8163) (which is... yeah ok bizarre :P). That doesn't really answer why Greg's implementation doesn't segfault and yours does but I know you're doing fancier things which may tickle .NET internals more.

But my attempt was to special-case .NET here:

https://github.com/bowtie-json-schema/bowtie/commit/dccfc421af1b6ae2dabee60605afe51692fb8958

and then to use .NET's native cross-compilation.

The remaining issue though is that the MSFT instructions assume you're using `docker`, rather than any other tool for building images (we happen to currently use `buildah`).

So I'm not 100% sure that route fixes the issue but it seems promising -- the remaining bit (especially if you want to give it a shot yourself) is figuring out what the equivalent of `TARGETARCH` is when you're building via `buildah bud`.

But that's all I figured out myself so far.

Reply:
(None of which was to contradict your issue by the way, that seems mayyyybe relevant too -- though hopefully the way your issue manifests isn't a segfault?!?)

Reply:
OK @mwadams I've got it nearly working on that branch I linked. (The relevant extra bit was that buildah on Ubuntu 22.04 was simply too old and we needed a newer one for TARGETARCH to work).

But now things build and... fail the smoke test. See e.g. https://github.com/bowtie-json-schema/bowtie/actions/runs/9158621357/job/25177365599

Unhelpfully we don't show error messages in `bowtie smoke`. But I'm pretty sure at this point it'll be something very simple (*famous last words*).

Reply:
Here's what it outputs:

https://github.com/bowtie-json-schema/bowtie/actions/runs/9158657283/job/25177471590#step:8:11

I think that error looks real and now might be a bug in the harness?

Reply:
Well, of course that's not the end of the story, as there's still some secondary bug now, as `bowtie smoke -i dotnet-corvus-jsonschema` still fails (and that reproduces locally), because the harness is sending valid results for `{"not": {}}` applied to instances (rather than invalid ones).

Reply:
OK that bug had to do with the retrieval URI that I guess your implementation requires (and which was hardcoded) -- but somehow you have some shared state there, and making that unique per-request fixes the issue.

See https://github.com/bowtie-json-schema/bowtie/commit/f07b66d22f114c919494b1a847c2427386cfed94

And now smoke finally passes -- so we can merge if you'd like, though I still see failures for 2020-12 tests, are you expecting those? Specifically looks like:

```
                          Bowtie                          
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┓
┃ Implementation             ┃ Skips ┃ Errors ┃ Failures ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━╇━━━━━━━━━━┩
│ Corvus.JsonSchema (dotnet) │ 0     │ 4      │ 20       │
└────────────────────────────┴───────┴────────┴──────────┘
```

The errors there are #703 -- does your implementation not ship copies of the metaschema internally?

The failures I haven't looked at.

Reply:
I'm not expecting errors.

I'll set it up to preload the metaschema. I don't preload it automatically because you only need it if you need it!

Reply:
(Actually that's not strictly true - I'm expecting errors for the datetime leap year tests.)

Reply:
I've taken your changes and added the metaschema as pre-requisites for the test suite.

Let's see what that does...

Reply:
Sorry you caught me at the exact moment in the middle of some other release-related changes, but think you've got them now.

Reply:
You pulled in the harness changes but not the .NET workflow special casing. Here, let me simply merge the branch I pointed you at, and then the additional changes can go on top.